owncloud/appManagement.js

```diff
@@ -53,9 +53,9 @@ apps.prototype.getApps = function() {
                     return;
                 }
 
-                if ((!(data in parser.xml2js(apps[0].body).ocs)) || (!(data in parser.xml2js(apps[1].body).ocs))) {
-                    reject(parser.xml2js(apps[0].body).ocs);
-                }
+                // if ((!(data in parser.xml2js(apps[0].body).ocs)) || (!(data in parser.xml2js(apps[1].body).ocs))) {
```

Comment:
why?

Reply:
data is undefined - no idea where data is supposed to come from.

deeper insights are welcome ....

Reply:
maybe it's supposed to be double quoted: `"data"` as the block seems to try and detect wheter "ocs.data" exists.

a simpler way to write this would be `if (parser.xml2js(apps[0].body).ocs.data)`

Reply:
probably not double quoted though... it does seem to be the equivalent of PHP's `isset`.

I suggest rewriting to the latter form